### PR TITLE
Port more widgets to fn3

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

File: sky/packages/sky/lib/src/fn3/animated_component.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          // We don't actually have any state to change, per se,
+          // we just know that we have in fact changed state.
+        });
+      });
+    }
+    performance.play(config.direction);
+  }
+
+  void didUpdateConfig(T oldConfig) {
+    // TODO(abarth): Don't we need to write config.direction into performance?
```
Comment (Hixie) [inline sky/packages/sky/lib/src/fn3/animated_component.dart]: that's what play() does

**Comment by Hixie** [comment]
Can any of the widgets/ tests be ported yet? Some of these widgets have tests, I think...

**Comment by Hixie** [comment]
LGTM

**Comment by abarth** [comment]
I'll work up to DatePicker, which has a bunch of tests.

**Pull request merged**